EndlessClient/Dialogs/BarberDialog.cs

```diff
@@ -0,0 +1,200 @@
+        private readonly IEODialogIconService _dialogIconService;
+        private readonly ILocalizedStringFinder _localizedStringFinder;
+        private readonly IBarberActions _barberActions;
+        private readonly ICharacterInventoryProvider _characterInventoryProvider;
+        private readonly IEOMessageBoxFactory _messageBoxFactory;
+        private readonly IEIFFileProvider _eifFileProvider;
+        private readonly ISfxPlayer _sfxPlayer;
+
+        private CharacterRenderProperties RenderProperties => _characterControl.RenderProperties;
+        private ListDialogItem _changeHairItem, _changeHairColor, _changeBuyHairStyleOrColor;
+
+        public BarberDialog(INativeGraphicsManager nativeGraphicsManager,
+                            ICharacterRendererFactory rendererFactory,
+                            IEODialogButtonService dialogButtonService,
+                            ICharacterRepository characterLvRepository,
+                            IEODialogIconService dialogIconService,
+                            ILocalizedStringFinder localizedStringFinder,
+                            IBarberActions barberActions,
+                            ICharacterInventoryProvider characterInventoryProvider,
+                            IEOMessageBoxFactory messageBoxFactory,
+                            IEIFFileProvider eifFileProvider,
+                            ISfxPlayer sfxPlayer)
+            : base(nativeGraphicsManager, isInGame: true)
+        {
+            _characterRepository = characterLvRepository;
+            _dialogIconService = dialogIconService;
+            _localizedStringFinder = localizedStringFinder;
+            _barberActions = barberActions;
+            _characterInventoryProvider = characterInventoryProvider;
+            _messageBoxFactory = messageBoxFactory;
+            _eifFileProvider = eifFileProvider;
+            _sfxPlayer = sfxPlayer;
+
+            BackgroundTexture = GraphicsManager.TextureFromResource(GFXTypes.PostLoginUI, 56);
+
+            var mainCharacterRenderProperties = _characterRepository.MainCharacter.RenderProperties;
+            _characterControl = new CreateCharacterControl(mainCharacterRenderProperties, rendererFactory)
+            {
+                DrawPosition = new Vector2(210, 19),
+            };
+
+            InitializeCharacterControl();
+            InitializeDialogItems(dialogButtonService);
+            CenterInGameView();
+        }
+
+        private void InitializeCharacterControl()
+        {
+            _characterControl.SetParentControl(this);
+        }
+
+        private void InitializeDialogItems(IEODialogButtonService dialogButtonService)
+        {
+            var cancel = CreateButton(dialogButtonService, new Vector2(215, 151), SmallButton.Cancel);
+            cancel.OnClick += (_, _) => Close(XNADialogResult.Cancel);
+
+            _changeHairItem = new ListDialogItem(this, ListDialogItem.ListItemStyle.Large, 0)
```

Comment:
Ok, last thing (probably):

- `OffsetY` is `25` by default for large style items. This does not need to be set here.
- Pass in `0`, `1`, and `2` as the list index for each subsequent list item (third constructor parameter, currently all zero). It automatically does the math for offsetting the list items (36*index for large style).
- Include `.DrawArea = ` and `.OffsetX = ` in the initializer lists

Reply:
You will end up with the incorrect data if you add include .DrawArea in the initializer lists as there's no method of pre-knowing the math's for the draw size.